### PR TITLE
Email notification when a role is changed/removed

### LINKED ISSUE #3264: Email notification for demoted/removed user
Followup to #1000: we should email the user (and possibly all the other owners/maintainers of a project as well) when they were an owner/maintainer of a project but have just been demoted or removed from the project.

> if a user is removed from a [maintainer or owner] role, then that user should also be emailed (e.g., to prevent an attacker who has compromised one owner, from silently removing other owners of a package, prior to uploading a new malicious package, thereby circumventing #997).

-- @edmorley

### PR DESCRIPTION
**Pull request by patelneel55**
Sends an email to the project owners and the associated user when a role is changed or removed. Closes #3264

### CODE REVIEW

File: warehouse/manage/views.py [hunk 1]
```diff
@@ -1595,6 +1627,26 @@ def delete_project_role(project, request):
...
+            owner_roles = (
+                request.db.query(Role)
+                .filter(Role.project == project)
+                .filter(Role.role_name == "Owner")
+                .all()
+            )
+            owner_users = {owner.user for owner in owner_roles}
+            send_collaborator_removed_email(
+                request,
+                owner_users,
```
Comment (di) [inline warehouse/manage/views.py:1640]:
```suggestion
                collaborators,
```

Comment (di) [reply warehouse/manage/views.py:1640]: I also think we want to do `collaborators.discard(role.user)` before we sent this as well.

File: warehouse/manage/views.py [hunk 2]
```diff
@@ -1548,6 +1552,34 @@ def change_project_role(project, request, _form_class=ChangeRoleForm):
...
                     },
                 )
+
+                owner_roles = (
+                    request.db.query(Role)
+                    .filter(Role.project == project)
+                    .filter(Role.role_name == "Owner")
+                    .all()
+                )
+                owner_users = {owner.user for owner in owner_roles}
```
[the remark this answers is not in the document]
Comment (patelneel55) [reply warehouse/manage/views.py:1562]: Currently, the `manage_project_roles` only sends emails to project owners. Do we also want that view to send emails to all collaborators?

[the remark this answers is not in the document]
Comment (di) [reply warehouse/manage/views.py:1562]: Thinking about this more: no, let's keep it just owners.

[the remark this answers is not in the document]
Comment (patelneel55) [reply warehouse/manage/views.py:1562]: I'm sorry, do you mean keep it owners for just `manage_project_roles` or for the rest of them as well?

[the remark this answers is not in the document]
Comment (di) [reply warehouse/manage/views.py:1562]: Just owners for all notifications. Non-owner collaborators don't have the ability to make these changes, so it makes sense (I think) to not notify them.

**Pull request merged**